### ISSUE TITLE
Cannot connect to the Docker daemon in Ubuntu

### ISSUE THREAD

**Issue by lpalmes** [drone-plugins/drone-docker#94]
It won't work in ubuntu 16.04, and gives me the error below.
I'm running a machine at digital ocean with this config `Ubuntu Docker 1.12.3 on 16.04`
And running drone:0.4 docker image.

```
[info] Pulling image plugins/drone-git:latest
Drone Git Plugin built from 43dcd64
$ git init
Initialized empty Git repository in /drone/src/gitlab.com/studiobox/amanger-backend/.git/
$ git remote add origin https://gitlab.com/studiobox/amanger-backend.git
$ git fetch --no-tags origin +refs/heads/master:
From https://gitlab.com/studiobox/amanger-backend
* branch            master     -> FETCH_HEAD
* [new branch]      master     -> origin/master
$ git reset --hard -q 78dbfd833602c8e6be897beb2271bc83ecc9e390
$ go build
[info] Pulling image plugins/drone-docker:latest
Drone Docker Plugin built at 2016-07-29 06:41:09 UTC
$ /usr/bin/docker daemon -g /drone/docker -s devicemapper
Cannot connect to the Docker daemon. Is the docker daemon running on this host?
Login failed.
[info] build failed (exit code 1)
```

Any help would be really aprecciated, and if some more information is needed let me know.
Thanks! And great project!

**Comment by donny-dont** [comment]
@lpalmes 0.5 is right around the corner. If you are just starting with Drone then you should use that.

**Comment by bradrydzewski** [comment]
@lpalmes device mapper is not a supported driver for this plugin. You can see more in the known issue section http://readme.drone.io/plugins/docker/

**Comment by lpalmes** [comment]
Oh, i used to do this in ubuntu 14, but know i couldn't get it to work, i tried also with done 0.5, but i got an error in the container logs like `version control not working`, that's why i went back to 0.4
Thanks for the help!

**Comment by bradrydzewski** [comment]
@lpalmes there isn't a `version control not working` not working, but there is a `version control system not configured` which happens when you try to start drone without correctly configuring the version control system (like github).

**Comment by lpalmes** [comment]
Oh yes, i couldn't remember exactly, but i'm using gitlab, in the dronerc file, the same one i use with 0.4, and i got that message.

**Comment by dgonzalez** [comment]
Heya, I got this problem as well and this is an absolute blocker for drone-docker on a private registry. I have tried pretty much everything that I've found in Internet but still does not work. Is this solved in Drone 0.5? What has to do the storage_driver with this problem? I don't see the correlation.

**Comment by tboerger** [comment]
The known problems and workarounds are documented within DOCS.md

**Comment by bradrydzewski** [comment]
@tboerger @dgonzales the troubleshooting section was moved to http://readme.drone.io/questions/why-does-the-docker-plugin-fail/

I am confident the troubleshooting guide will help you resolve the issue. If not, we will need more detail to point you in the right direction. Right now we do not have adequate information (logs, error message, etc) to help you.

> this is an absolute blocker for drone-docker on a private registry

using the plugin with a private registry works perfectly fine. There is an example in the documentation dedicated to private registries. See http://plugins.drone.io/drone-plugins/drone-docker/

> Is this solved in Drone 0.5?

There is no issue to solve, that I'm aware of.

> What has to do the storage_driver with this problem

This is somewhat discussed in the troubleshooting docs, linked above.

This is because the plugin runs docker in docker for isolation purposes, and certain storage drivers do not work when running docker in docker. The devicemapper drivers is one of those drivers.
